test/support/factory.ex

```diff
@@ -187,21 +187,29 @@ defmodule Wanda.Factory do
   def check_results_from_targets_factory(attrs) do
     targets = Map.get(attrs, :targets, [])
     result = Map.get(attrs, :result, :passing)
+    failure_message = Map.get(attrs, :failure_message)
 
     targets
     |> Enum.flat_map(& &1.checks)
     |> Enum.uniq()
     |> Enum.map(fn check_id ->
       check_targets = Enum.filter(targets, &(check_id in &1.checks))
-      check_result_from_target(check_id, check_targets, result)
+      check_result_from_target(check_id, check_targets, result, failure_message)
     end)
   end
 
-  defp check_result_from_target(check_id, check_targets, result) do
+  defp check_result_from_target(check_id, check_targets, result, failure_message) do
```

Comment:
The failure message will be a little bit dump, as it puts the same message to all the checks. Maybe we could use `sequence` or similar.
But it is fine as it is by now. We could improve if needed.